### ISSUE TITLE
Possible NPE

### ISSUE THREAD

**Issue by carmudgeon** [SparkPost/java-sparkpost#44]
i got this today:

`java.lang.Thread.run(Thread.java:745) Caused by: java.lang.NullPointerException at com.sparkpost.transport.RestConnection.receiveResponse(RestConnection.java:226) at com.sparkpost.transport.RestConnection.doHttpMethod(RestConnection.java:279) at com.sparkpost.transport.RestConnection.post(RestConnection.java:322) at com.sparkpost.resources.ResourceTransmissions.create(ResourceTransmissions.java:28) at `

**Comment by yepher** [comment]
Do you know how to reproduce that?

Are you seeing it often or has it only happened once?

**Comment by carmudgeon** [comment]
At least 4 times, and no, i don't have the way to reproduce it.

**Comment by yepher** [comment]
Ok, thanks for reporting it. I will look into it.

**Comment by yepher** [comment]
@carmudgeon,

Are you still seeing this NPE? I think I've narrowed down the source of this error some but want to see if you are still seeing the problem.

**Comment by carmudgeon** [comment]
i occurred twice yesterday 27/04/2016 same stacktrace, i managed to catch it, but the response will be null so i will not be able to tell if the email was sent or not.

**Comment by yepher** [comment]
Thanks for the update. I'm trying to understand why this happen. The code change will be to throw a SparkPost exception like it is trying to do anyway but did not just want to add the catch exception quite yet until I can understand it better. I am expecting this is some sort of connection error or the socket is getting dropped for some reason.

I will get a fix out soon.

**Comment by kalnik-a-a** [comment]
@yepher 
>  I am expecting this is some sort of connection error or the socket is getting dropped for some reason.

I believe this error can be reproduced in various ways. 
Simple way to reproduce:
```
RestConnection restConnection = new RestConnection(new Client("YOUR_API_KEY"));
Webhook webhook = new Webhook();
webhook.setName("name with spaces");
ResourceWebhooks.update(restConnection, webhook.getName(), webhook);
```
Reason: server returns empty response. In my case it was response with 400 code and empty content( conn.getContentLength() == 0).

Btw, ResourceWebhooks.update must use PUT instead of POST =)

**Comment by kalnik-a-a** [comment]
One more way: delete existing webhook. Response with OK status code and empty body. NPE was thrown.

**Comment by carmudgeon** [comment]
In my case, it was during a transmission

**Comment by yepher** [comment]
Thank you trying to get to this today.

**Comment by yepher** [comment]
I've put a fix in a branch

**Comment by carmudgeon** [comment]
Do you know when is this going to be merged?

**Comment by yepher** [comment]
It is merged. Just need to finish testing and put in Maven Repo. I will release this week.

**Comment by yepher** [comment]
Fixed in v0.16.1
